awpy/analytics/map_control.py

```diff
@@ -1,330 +1,368 @@
-
-
-def approx_neighbors(
+from awpy.types import (
+    BFSTileData,
+    FrameMapControlValues,
+    FrameTeamMetadataDict,
+    FrameTeamMetadataObject,
+    GameFrame,
+    GameRound,
+    TeamFrameInfo,
+    TeamMapControlValuesDict,
+    TileDistanceObject,
+    TileIdObject,
+    TileNeighborsDict,
+)
+
+
+def _approx_neighbors(
     map_name: str,
-    source_tile_id: int,
-    n: int = 5,
-) -> list[tuple[float, int]]:
+    source_tile_id: TileIdObject,
+    n_neighbors: int = 5,
+) -> list[TileDistanceObject]:
     """Approximates neighbors for isolated tiles by finding n closest tiles.
 
     Args:
-        map_name (string)   : Map for source_tile_id
-        source_tile_id (int): Area ID as an integer
-        n (int)             : Number of closest tiles/approximated
-                              neighbors wanted
+        map_name (str): Map for source_tile_id
+        source_tile_id (TileIdObject): TileIdObject for source tile
+        n_neighbors (int): Number of closest tiles/approximated neighbors wanted
+
     Returns:
-        List of tuples representing n closest tiles.
-            Tuple[0] -> Distance between source_tile_id and represented tile
-            Tuple[1] -> Area ID of represented tile
+        List of TileDistanceObjects for n closest tiles
+
     Raises:
-        ValueError: If map_name is not in awpy.data.NAV
-                    If source_tile_id is not in awpy.data.NAV[map_name]
-                    If the length of point is not 3
+        ValueError: If source_tile_id is not in awpy.data.NAV[map_name]
+                    If n_neighbors <= 0
     """
-    if map_name not in NAV:
-        raise ValueError("Map not found.")
     if source_tile_id not in NAV[map_name].keys():
-        raise ValueError("Area ID not found.")
-    if n < 0:
-        raise ValueError("n must be >= 0")
+        msg = "Area ID not found."
+        raise ValueError(msg)
+    if n_neighbors <= 0:
+        msg = "Invalid n_neighbors value. Must be > 0"
+        raise ValueError(msg)
 
     current_map_info = NAV[map_name]
-    distance_arr: list[tuple[float, int]] = []
+    possible_neighbors_arr: list[TileDistanceObject] = []
 
     for tile in current_map_info:
         if tile != source_tile_id:
+            # TODO: See if AREA_DIST_MATRIX is worth incorporating
             current_distance = area_distance(
                 map_name, tile, source_tile_id, dist_type="euclidean"
             )["distance"]
-            distance_arr.append((current_distance, tile))
-    distance_arr.sort()
+            current_tile_id = TileIdObject(tile)
+            current_tile_distance_obj: TileDistanceObject = {
+                "distance": current_distance,
+                "tile_id": current_tile_id,
+            }
+            possible_neighbors_arr.append(current_tile_distance_obj)
+
+    sorted_neighbors = sorted(possible_neighbors_arr, key=lambda d: d["distance"])
 
-    return distance_arr[:n]
+    return sorted_neighbors[:n_neighbors]
```

Comment:
This can also be one line `return sorted(possible_neighbors_arr, key=lambda d: d["distance"])[:n_neighbors]`